### PR TITLE
should log error when error in parsing device plugin image

### PR DESCRIPTION
**Pull request by tianshapjq**
**What this PR does / why we need it**:
add some extra error log in parsing the device plugin image. Error happened in my setup when fetching the yaml from url, but since no error log printed it cost a long time to figure out the reason. So it'd be nice to print the error IMO.

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

/sig-node

### CODE REVIEW

**Comment by RenaudWasTaken** [comment]
cc @mindprince @jiayingz 
/lgtm

File: test/e2e_node/image_list.go
Comment (mindprince) [inline test/e2e_node/image_list.go]: Can you log the errors in `GetGPUDevicePluginImage()` itself instead of returning it and then logging it here?

Comment (tianshapjq) [reply test/e2e_node/image_list.go]: @mindprince sure. I was just afraid that would import a new package in the gpu_list.go, which I'm not sure if causes the loop of imports.

**Comment by tianshapjq** [comment]
@mindprince new updates. PATL~

**Review by rohitagarwal003 — APPROVED** [review]
/lgtm

**Comment by tianshapjq** [comment]
@mindprince @RenaudWasTaken can be be lgtm and merged?

File: test/e2e/framework/gpu_util.go
Comment (RenaudWasTaken) [inline test/e2e/framework/gpu_util.go]: Any specific reason why no containers needs to be checked? Wouldn't DsFromManifest return that in the err field?

**Comment by RenaudWasTaken** [comment]
I don't have approuver status but /lgtm

**Comment by RenaudWasTaken** [comment]
/lgtm
I don't have approuver status

File: test/e2e/framework/gpu_util.go
Comment (tianshapjq) [reply test/e2e/framework/gpu_util.go]: Actually I'm not sure about the specific reason, maybe pre-checking would return error ASAP.

**Comment by tianshapjq** [comment]
@vishh can this be approved?

**Comment by tpepper** [comment]
@tianshapjq @vishh @RenaudWasTaken @mindprince when reviewing and approving please consider improving the commit message quality for future code maintainability.  Two commits with just "update" is less than informative when seen in the git log in isolation.  This series of patches also would have better been squashed before merge.

**Pull request merged**
